### Q1: Write a python program to scrape data for “Data Analyst” Job position in “Bangalore” location. You have to scrape the job-title, job-location, company_name, experience_required. You have to scrape first 10 jobs data. This task will be done in following steps:


In [4]:
import pandas as pd
import time
import warnings
warnings.filterwarnings('ignore')
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.common.exceptions import NoSuchElementException

In [2]:
# connecting Webdriver
driver = webdriver.Chrome(r'C:\chromedriver.exe')
driver

C:\Users\SURAJ\AppData\Local\Temp\ipykernel_7648\2650290596.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(r'C:\chromedriver.exe')


<selenium.webdriver.chrome.webdriver.WebDriver (session="dc04ab107b200d171647f764829aa4f2")>

In [3]:
# Opening naukri.com
url ='https://www.naukri.com/'
driver.get(url)

In [5]:
designation = driver.find_element(By.CLASS_NAME,"suggestor-input ")
designation.send_keys('Data Analyst') 

In [7]:
location = driver.find_element(By.XPATH,"/html/body/div[1]/div[6]/div/div/div[5]/div/div/div/div[1]/div/input")
location.send_keys('Bengalore')

In [8]:
search = driver.find_element(By.CLASS_NAME,"qsbSubmit")
search.click()

In [9]:
job_title =[]
job_location=[]
company_name =[]
experience_required =[]

In [10]:
title_tags=driver.find_elements(By.XPATH,'//a[@class="title ellipsis"]')
for i in title_tags [0:10]:
    title=i.text
    job_title.append(title)

In [11]:
location_tags=driver.find_elements(By.XPATH,'//span[@class="ellipsis fleft locWdth"]')
for i in location_tags[0:10]:
    location=i.text
    job_location.append(location)

In [12]:
company_tags=driver.find_elements(By.XPATH,'//a[@class="subTitle ellipsis fleft"]')
for i in company_tags[0:10]:
    company=i.text
    company_name.append(company)

In [13]:
experience_tags=driver.find_elements(By.XPATH,'//span[@class="ellipsis fleft expwdth"]')
for i in experience_tags[0:10]:
    exp=i.text
    experience_required.append(exp)

In [14]:
print(len(job_title), len(job_location), len(company_name), len(experience_required))

10 10 10 10


In [15]:
import pandas as pd
df=pd.DataFrame({'Title': job_title, 'Location': job_location,'Company Name': company_name,'Experience': experience_required})
df

Title  \
0                               Data Analyst   
1                               Data Analyst   
2                               Data Analyst   
3  Environmental Sustainability Data Analyst   
4                               Data Analyst   
5                               Data Analyst   
6                      Data Analyst - EdTech   
7                               Data Analyst   
8                 Data Engineer/Data Analyst   
9                 Data Analyst - Contractual   

                                            Location  \
0                                Bangalore/Bengaluru   
1                                Bangalore/Bengaluru   
2                                Bangalore/Bengaluru   
3  Kolkata, Mumbai, New Delhi, Hyderabad/Secunder...   
4                                Bangalore/Bengaluru   
5                                Bangalore/Bengaluru   
6                                Bangalore/Bengaluru   
7                                Bangalore/Bengaluru   
8  Hybrid - Kolkata, Hyderabad/Secunderabad, Pune...   
9                                Bangalore/Bengaluru   

                   Company Name Experience  
0                  Globals Ites    2-3 Yrs  
1                        Varite    4-5 Yrs  
2  Cynosure Corporate Solutions    2-7 Yrs  
3             PerkinElmer, Inc.    2-6 Yrs  
4                NNE Pharmaplan    1-2 Yrs  
5                Ingersoll Rand    3-6 Yrs  
6                   Talentstack    2-6 Yrs  
7                  Unusual Hire    1-4 Yrs  
8                 Tech Mahindra   6-11 Yrs  
9  Search Advisers Services Guj    2-3 Yrs

### Q2: Write a python program to scrape data for “Data Scientist” Job position in “Bangalore” location. You have to scrape the job-title, job-location,company_name, full job-description. You have to scrape first 10 jobs data.

In [16]:
import pandas as pd
import time
import warnings
warnings.filterwarnings('ignore')
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.common.exceptions import NoSuchElementException

In [17]:
driver = webdriver.Chrome(r'C:\chromedriver.exe')
driver

<selenium.webdriver.chrome.webdriver.WebDriver (session="c7d76144c35f3974670cacf2d095e877")>

In [18]:
url ='https://www.naukri.com/'
driver.get(url)

In [19]:
designation = driver.find_element(By.CLASS_NAME,"suggestor-input ")
designation.send_keys('Data Scientist')

In [21]:
location = driver.find_element(By.XPATH,"/html/body/div[1]/div[6]/div/div/div[5]/div/div/div/div[1]/div/input")
location.send_keys('Bengalore')

In [22]:
search = driver.find_element(By.CLASS_NAME,"qsbSubmit")
search.click()

In [23]:
job_title =[]
job_location=[]
company_name =[]
experience_required =[]

In [24]:
title_tags=driver.find_elements(By.XPATH,'//a[@class="title ellipsis"]')
for i in title_tags [0:10]:
    title=i.text
    job_title.append(title)

In [25]:
location_tags=driver.find_elements(By.XPATH,'//span[@class="ellipsis fleft locWdth"]')
for i in location_tags[0:10]:
    location=i.text
    job_location.append(location)

In [26]:
company_tags=driver.find_elements(By.XPATH,'//a[@class="subTitle ellipsis fleft"]')
for i in company_tags[0:10]:
    company=i.text
    company_name.append(company)

In [27]:
experience_tags=driver.find_elements(By.XPATH,'//span[@class="ellipsis fleft expwdth"]')
for i in experience_tags[0:10]:
    exp=i.text
    experience_required.append(exp)

In [28]:
print(len(job_title), len(job_location), len(company_name), len(experience_required))

10 10 10 10


In [29]:
import pandas as pd
df=pd.DataFrame({'Title': job_title, 'Location': job_location,'Company Name': company_name,'Experience': experience_required})
df

Title  \
0  Data Science Professional - IBM SPSS Statistic...   
1                            Data Science Specialist   
2                   Analystics & Modeling Specialist   
3                              Senior data scientist   
4                                 Data Scientist_NLP   
5                                     Data Scientist   
6                    Machine Learning (AI) Architect   
7                                 Sr. Data scientist   
8       Manager - Innovations Hub - Machine Learning   
9                                Lead Data Scientist   

                                            Location           Company Name  \
0  Noida, Mumbai, Pune, Chennai, Bangalore/Bengaluru  Hexaware Technologies   
1  Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...              Accenture   
2  Kolkata, Mumbai, Hyderabad/Secunderabad, Pune,...              Accenture   
3  Mumbai, Pune, Chennai, Gurgaon/Gurugram, Banga...      Fractal Analytics   
4  Mumbai, Pune, Chennai, Gurgaon/Gurugram, Banga...      Fractal Analytics   
5  Mumbai, Pune, Chennai, Gurgaon/Gurugram, Banga...      Fractal Analytics   
6  Kolkata, Mumbai, New Delhi, Hyderabad/Secunder...             Persistent   
7   Hybrid - Bangalore/Bengaluru, Mumbai (All Areas)                   PayU   
8  Mumbai, Hyderabad/Secunderabad, Bangalore/Beng...                    PwC   
9                                Bangalore/Bengaluru                Genpact   

  Experience  
0    5-8 Yrs  
1    2-4 Yrs  
2    6-8 Yrs  
3    4-8 Yrs  
4   5-11 Yrs  
5    3-7 Yrs  
6   5-12 Yrs  
7    1-6 Yrs  
8    7-9 Yrs  
9   8-12 Yrs

### Q3: In this question you have to scrape data using the filters available on thewebpage as shown below: You have to use the location and salary filter. You have to scrape data for “Data Scientist” designation for first 10 job results. You have to scrape the job-title, job-location, company_name, experience_required. The location filter to be used is “Delhi/NCR” . The salary filter to be used is “3-6” lakhs.

In [30]:
import pandas as pd
import time
import warnings
warnings.filterwarnings('ignore')
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.common.exceptions import NoSuchElementException

In [31]:
driver = webdriver.Chrome(r'C:\chromedriver.exe')
driver

<selenium.webdriver.chrome.webdriver.WebDriver (session="06a00571a5729d6d465e0426935776e3")>

In [32]:
url ='https://www.naukri.com/'
driver.get(url)

In [33]:
designation = driver.find_element(By.CLASS_NAME,"suggestor-input ")
designation.send_keys('Data Scientist')

In [34]:
location = driver.find_element(By.XPATH,"/html/body/div[1]/div[6]/div/div/div[5]/div/div/div/div[1]/div/input")
location.send_keys('Delhi/NCR')

In [35]:
search = driver.find_element(By.CLASS_NAME,"qsbSubmit")
search.click()

In [36]:
job_title =[]
job_location=[]
company_name =[]
experience_required =[]

In [37]:
title_tags=driver.find_elements(By.XPATH,'//a[@class="title ellipsis"]')
for i in title_tags [0:10]:
    title=i.text
    job_title.append(title)

In [38]:
location_tags=driver.find_elements(By.XPATH,'//span[@class="ellipsis fleft locWdth"]')
for i in location_tags[0:10]:
    location=i.text
    job_location.append(location)

In [39]:
company_tags=driver.find_elements(By.XPATH,'//a[@class="subTitle ellipsis fleft"]')
for i in company_tags[0:10]:
    company=i.text
    company_name.append(company)

In [40]:
experience_tags=driver.find_elements(By.XPATH,'//span[@class="ellipsis fleft expwdth"]')
for i in experience_tags[0:10]:
    exp=i.text
    experience_required.append(exp)

In [41]:
salary_filter = driver.find_element(By.XPATH, '//span[@title="3-6 Lakhs"]')
salary_filter.click()

In [42]:
print(len(job_title), len(job_location), len(company_name), len(experience_required))

10 10 10 10


In [43]:
import pandas as pd
df=pd.DataFrame({'Title': job_title, 'Location': job_location,'Company Name': company_name,'Experience': experience_required})
df

Title  \
0  Data Science Professional - IBM SPSS Statistic...   
1                              Senior data scientist   
2                                 Data Scientist_NLP   
3                                     Data Scientist   
4                    Machine Learning (AI) Architect   
5                                     Data Scientist   
6                                     Data Scientist   
7                                       Data Science   
8                                     Data Scientist   
9                              Senior Data Scientist   

                                            Location           Company Name  \
0  Noida, Mumbai, Pune, Chennai, Bangalore/Bengaluru  Hexaware Technologies   
1  Mumbai, Pune, Chennai, Gurgaon/Gurugram, Banga...      Fractal Analytics   
2  Mumbai, Pune, Chennai, Gurgaon/Gurugram, Banga...      Fractal Analytics   
3  Mumbai, Pune, Chennai, Gurgaon/Gurugram, Banga...      Fractal Analytics   
4  Kolkata, Mumbai, New Delhi, Hyderabad/Secunder...             Persistent   
5  Temp. WFH - Kochi/Cochin, Kolkata, Hyderabad/S...              Cognizant   
6  Hyderabad/Secunderabad, Gurgaon/Gurugram, Bang...          Tech Mahindra   
7  Kolkata, Mumbai, New Delhi, Hyderabad/Secunder...               Skillety   
8  Kolkata, Mumbai, New Delhi, Hyderabad/Secunder...               Clarifai   
9  Kolkata, Mumbai, New Delhi, Hyderabad/Secunder...        Corporate Steps   

  Experience  
0    5-8 Yrs  
1    4-8 Yrs  
2   5-11 Yrs  
3    3-7 Yrs  
4   5-12 Yrs  
5   6-10 Yrs  
6   5-10 Yrs  
7   7-10 Yrs  
8    3-7 Yrs  
9   5-10 Yrs

### Q4: Write a python program to scrape data for first 10 job results for Data scientist 
Designation in Noida location. You have to scrape company_name, No. of days 
ago when job was posted, Rating of the company.

In [44]:
import selenium
import pandas as pd
from selenium import webdriver
import time
from selenium.common.exceptions import NoSuchElementException

# connecting Webdriver
driver = webdriver.Chrome(r'C:\chromedriver.exe')
driver

# Opening website
url ='https://www.glassdoor.co.in/index.htm'
driver.get(url)

In [45]:
import requests

In [46]:
page=requests.get("https://www.glassdoor.co.in/index.htm")
page

<Response [403]>

###### we are getting response 403 . so we can't scrape the data from this website.

#### Q5: Write a python program to scrape the salary data for Data Scientist designation¶
in Noida .

In [47]:
import requests

In [48]:
page=requests.get("https://www.glassdoor.co.in/index.htm")
page

<Response [403]>

###### we are getting response 403 . so we can't scrape the data from this website.

##### Q6 : Scrape data of first 100 sunglasses listings on flipkart.com. You have to 
scrape four attributes:


In [49]:
import pandas as pd
import time
import warnings
warnings.filterwarnings('ignore')
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.common.exceptions import NoSuchElementException

In [50]:
driver = webdriver.Chrome(r'C:\chromedriver.exe')
driver

<selenium.webdriver.chrome.webdriver.WebDriver (session="d18d3c62a2efc30d926758cbf1fb9a74")>

In [51]:
url ='https://www.flipkart.com/'
driver.get(url)

In [52]:
pop_up = driver.find_element("xpath","//div[@class='_2QfC02']/button[@class = '_2KpZ6l _2doB4z']")
pop_up.click()

In [53]:
prod_ser = driver.find_element("xpath","//div[@class='_3OO5Xc']//input[@type = 'text']")
prod_ser.send_keys("sunglasses")

In [54]:
sub_button = driver.find_element("xpath","//div[@class = 'col-12-12 _2oO9oE']//button[@class = 'L0Z3Pu']")
sub_button.click()

In [55]:
#creating the empty lists
Brand=[]
Product_Description=[]
Price=[]
Discount=[]

In [56]:
import time

# use time.sleep() to pause the program for a specified number of seconds
time.sleep(3)

In [58]:
URL = []
for i in range(0, 4):
    sunglasses = driver.find_elements("xpath", "//a[@class='_1jJQdf _2Mji8F']")
    for s in sunglasses:
        URL.append(s.get_attribute('href'))
    next_button = driver.find_element("xpath", '//a[@class="_1LKTO3"]')
    next_button.click()
    time.sleep(3)



In [59]:
URL

['https://www.flipkart.com/sunglasses/pr?sid=26x&q=sunglasses&otracker=categorytree',
 'https://www.flipkart.com/sunglasses/pr?sid=26x&q=sunglasses&otracker=categorytree',
 'https://www.flipkart.com/sunglasses/pr?sid=26x&q=sunglasses&otracker=categorytree',
 'https://www.flipkart.com/sunglasses/pr?sid=26x&q=sunglasses&otracker=categorytree']

In [61]:
list_of_url=URL[0:100]

In [63]:
#running a loop to extract all the required information from the list_of_url
from tqdm import tqdm
for i in tqdm(list_of_url):
    driver.get(i)
    try:
        brand=driver.find_element("xpath","//span[@class='_2WkVRV']").text
        Brand.append(brand)
        description=driver.find_element("xpath","//span[@class='_2B099V']").text
        Product_Description.append(description)
        price=driver.find_element("xpath","//div[@class='_30jeq3']").text
        Price.append(price)
        discount=driver.find_element("xpath","//div[@class='_3Ay6Sb']").text
        Discount.append(discount)
    except NoSuchElementException:
        Brand.append("Null")
        Product_Description.append("Null")
        Price.append("Null")
        Discount.append('Null')
    time.sleep(3)

100%|████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:24<00:00,  6.23s/it]


In [64]:
print(len(Brand),len(Product_Description),len(Price),len(Discount))

4 4 4 4


In [65]:
import pandas as pd
df = pd.DataFrame()

In [66]:
sunglasses=pd.DataFrame({})
sunglasses['Brand']=Brand[:100]
sunglasses['Product_Description']=Product_Description[:100]
sunglasses['Price']=Price[:100]
sunglasses['Discount']=Discount[:100]
sunglasses

Brand Product_Description Price Discount
0  Null                Null  Null     Null
1  Null                Null  Null     Null
2  Null                Null  Null     Null
3  Null                Null  Null     Null

##### Q7: Scrape 100 reviews data from flipkart.com for iphone11 phone. You have to 
go the link: